dist/index.js

```diff
@@ -77,14 +77,31 @@ function run() {
             /**
              * Get all commits on the PR.
              */
-            core.info(`Getting commits for PR number ${pullRequest.number}...`);
-            const response = yield octokit.request("GET /repos/{owner}/{repo}/pulls/{pull_number}/commits", Object.assign(Object.assign({}, github.context.repo), { pull_number: pullRequest.number, per_page: 100 }));
-            core.debug(JSON.stringify(response));
-            core.info(`Found ${response.data.length} commits:`);
+            let commits = [];
+            /**
+             * Attempt to get all commits on the PR. This endpoint will return a maximum of
+             * 250 commits, with a 100 per page limit. Loop over pages 1, 2 and 3 to create an
+             * array of commits.
+             */
+            [0, 100, 200].forEach((max, index) => __awaiter(this, void 0, void 0, function* () {
```

Comment:
instead of a foreach can't we use a while?

```
const per_page=100;
const max_results=300;

while(commits.length < max_results && commits.length % per_page == 0){ ... }

```

Reply:
If a PR had 100 or 200 commits exactly, this would get stuck in an infinite loop.

Reply:
Indeed it will I knew I must've gotten something wrong. Thanks for the review of the review hahaha

Reply:
I also need the `index` of each iteration, as that's what I use to request the next paged result.

For that reason I think `forEach` is the better choice :)